Alvaro Gaming/PythonGaming_p3.py

```diff
@@ -14,79 +14,114 @@
 DISPLAYSURF.fill(WHITE)
-pygame.display.set_caption("Alvy Game")
+pygame.display.set_caption("Alvy Game: Pong")
+
 
 class Enemy(pygame.sprite.Sprite):
     def __init__(self):
         super().__init__()
         self.image = pygame.image.load("Enemy.png")
         self.rect = self.image.get_rect()
-        self.rect.center = (random.randint(40, SCREEN_WIDTH-40), 0)
+        #self.rect.center = (random.randint(40, SCREEN_WIDTH-40), 0)
+
+        self.velocity = [random.randint(4, 8), random.randint(-8, 8)]
 
+    """""
     def move(self):
-        self.rect.move_ip(0,10)
+        self.rect.move_ip(0, 10)
         if self.rect.bottom > 600:
             global SCORE
             SCORE += 1
             self.rect.top = 0
             self.rect.center = (random.randint(30, 370), 0)
+            
+    """
 
-    #This line of code is now shorten and done using sprite groups
-    #def draw(self, surface):
-        #surface.blit(self.image, self.rect)
+    def move(self):
+        self.rect.x += self.velocity[0]
+        self.rect.y += self.velocity[1]
 
+    def bounce(self):
+        self.velocity[0] = -self.velocity[0]
+        self.velocity[1] = random.randint(-8, 8)
 
-class Player(pygame.sprite.Sprite):
-    def __init__(self):
+    # This line of code is now shorten and done using sprite groups
+    # def draw(self, surface):
+    # surface.blit(self.image, self.rect)
+
+
+class Player_1(pygame.sprite.Sprite):
+    def __init__(self, coord):
         super().__init__()
         self.image = pygame.image.load("Player.png")
         self.rect = self.image.get_rect()
-        self.rect.center = (160, 520)
+        self.rect.center = coord
 
     def move(self):
         pressed_keys = pygame.key.get_pressed()
-        if pressed_keys[K_UP]:
+        if pressed_keys[K_UP] and self.rect.top > 0:
             self.rect.move_ip(0, -5)
-        if pressed_keys[K_DOWN]:
+        if pressed_keys[K_DOWN] and self.rect.bottom < 600:
             self.rect.move_ip(0, 5)
 
-        if self.rect.left > 0:
-            if pressed_keys[K_LEFT]:
-                self.rect.move_ip(-5, 0)
-        if self.rect.right < SCREEN_WIDTH:
-            if pressed_keys[K_RIGHT]:
-                self.rect.move_ip(5, 0)
+
+class Player_2(pygame.sprite.Sprite):
+    def __init__(self, coord):
+        super().__init__()
+        self.image = pygame.image.load("Player.png")
+        self.rect = self.image.get_rect()
+        self.rect.center = coord
+
+    def move(self):
+        pressed_keys = pygame.key.get_pressed()
+        if pressed_keys[K_w] and self.rect.top > 0:
+            self.rect.move_ip(0, -5)
+        if pressed_keys[K_s] and self.rect.bottom < 600:
+            self.rect.move_ip(0, 5)
+
+        # if self.rect.left > 0:
+        # if pressed_keys[K_LEFT]:
+        # self.rect.move_ip(-5, 0)
+        # if self.rect.right < SCREEN_WIDTH:
+        # if pressed_keys[K_RIGHT]:
+        # self.rect.move_ip(5, 0)
+
+
+st_cord_1 = (35, 520)
+st_cord_2 = (765, 100)
```

Comment:
st? not sure what the naming is supposed to mean, maybe scoreboard?

edit: I see it's the player "starting" coordinates. Better naming would be helpful.

Reply:
fudge, you are right, even I forgot what I meant by st lololol. I didn't want to make the name too long and thought it smart to do it short but ended up backfiring haha

what is too long of a variable/constant name?

Reply:
It's mainly about being clear and readable, so that's the priority over length